core/domain/html_cleaner.py

```diff
@@ -191,3 +192,371 @@ def get_rte_components(html_string: str) -> List[ComponentsDict]:
+            )
+
+        caption_value = utils.unescape_html(
+                tag['caption-with-value'])[1:-1].replace('\\"', '')
+        if len(caption_value.strip()) > 500:
+            raise utils.ValidationError(
+                'Image tag \'caption-with-value\' attribute should not '
+                'be greater than 500 characters.'
+            )
+
+        if not tag.has_attr('filepath-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'filepath-with-value\' attribute.'
+            )
+
+        filepath_value = utils.unescape_html(
+                tag['filepath-with-value'])[1:-1].replace('\\"', '')
+        if filepath_value.strip() in empty_values:
+            raise utils.ValidationError(
+                'Image tag \'filepath-with-value\' attribute should not '
+                'be empty.'
+            )
+
+    for tag in soup.find_all('oppia-noninteractive-skillreview'):
+        if not tag.has_attr('text-with-value'):
+            raise utils.ValidationError(
+                'SkillReview tag does not have \'text-with-value\' '
+                'attribute.'
+            )
+
+        text_value = utils.unescape_html(
+                tag['text-with-value'])[1:-1].replace('\\"', '')
+        if text_value.strip() in empty_values:
+            raise utils.ValidationError(
+                'SkillReview tag should not have \'text-with-value\' '
+                'attribute value as empty string.'
+            )
+
+        if not tag.has_attr('skill_id-with-value'):
+            raise utils.ValidationError(
+                'SkillReview tag does not have \'skill_id-with-value\' '
+                'attribute.'
+            )
+
+        skill_value = utils.unescape_html(
+                tag['skill_id-with-value'])[1:-1].replace('\\"', '')
+        if skill_value.strip() in empty_values:
+            raise utils.ValidationError(
+                'SkillReview tag should not have '
+                '\'skill_id-with-value\' attribute value as empty string.'
+            )
+
+    for tag in soup.find_all('oppia-noninteractive-video'):
+        if not tag.has_attr('start-with-value'):
+            raise utils.ValidationError(
+                'Video tag does not have \'start-with-value\' '
+                'attribute.'
+            )
+
+        if tag['start-with-value'].strip() in empty_values:
+            raise utils.ValidationError(
+                'Video tag \'start-with-value\' attribute should not '
+                'be empty.'
+            )
+
+        if not tag.has_attr('end-with-value'):
+            raise utils.ValidationError(
+                'Video tag does not have \'end-with-value\' '
+                'attribute.'
+            )
+
+        if tag['end-with-value'].strip() in empty_values:
+            raise utils.ValidationError(
+                'Video tag \'end-with-value\' attribute should not '
+                'be empty.'
+            )
```

Comment:
These three can be parametrized and put into a function, then you can reuse the function in a few places.

Reply:
Done